hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/client/checksum/CompositeCrc.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.client.checksum;
+
+import com.google.common.base.Preconditions;
+import org.apache.hadoop.fs.CompositeCrcFileChecksum;
+import org.apache.hadoop.fs.FileChecksum;
+import org.apache.hadoop.fs.Options;
+import org.apache.hadoop.hdds.client.ECReplicationConfig;
+import org.apache.hadoop.hdds.protocol.datanode.proto.ContainerProtos;
+import org.apache.hadoop.hdds.protocol.datanode.proto.ContainerProtos.ChecksumType;
+import org.apache.hadoop.hdds.protocol.datanode.proto.ContainerProtos.ChunkInfo;
+import org.apache.hadoop.io.DataOutputBuffer;
+import org.apache.hadoop.ozone.HadoopCompatibility;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyLocationInfo;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.ChecksumTypeProto;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.CompositeCrcFileChecksumProto;
+import org.apache.hadoop.util.CrcComposer;
+import org.apache.hadoop.util.CrcUtil;
+import org.apache.hadoop.util.DataChecksum;
+import org.apache.ratis.thirdparty.com.google.protobuf.ByteString;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import static org.apache.ratis.util.Preconditions.assertInstanceOf;
+
+/**
+ * Hides usage of Hadoop-version specific CompositeCrcFileChecksum and its helper classes.
+ * Before accessing methods of this class, callers should check if
+ * {@link HadoopCompatibility#isCompositeCrcAvailable()}.
+ *
+ * <pre>
+ *   if (HadoopCompatibility.isAvailable()) {
+ *     CompositeCrc.&lt;methodToRun(...)&gt;
+ *   }
+ * </pre>
+ */
+public final class CompositeCrc {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(CompositeCrc.class);
+
+  private CompositeCrc() {
+    // no instances
+  }
+
+  public static void checkAvailability() {
+    LOG.info("{} is available", CompositeCrcFileChecksum.class.getSimpleName());
```

Comment:
IMO better move this one to debug level otherwise too much noise.

Reply:
Updated to `debug`.  To keep log levels consistent, I also reduced this one to `trace`:

https://github.com/apache/ozone/blob/c3a15b3010314fb9039fc69c85de500ab877bd10/hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/client/checksum/CompositeCrc.java#L87-L97